pkg/api/util/conditions.go

```diff
@@ -96,17 +96,18 @@ func SetIssuerCondition(i cmapi.GenericIssuer, observedGeneration int64, conditi
 }
 
 // CertificateHasCondition will return true if the given Certificate has a
-// condition matching the provided CertificateCondition.
-// Only the Type and Status field will be used in the comparison, meaning that
-// this function will return 'true' even if the Reason, Message and
+// condition matching the provided CertificateCondition with a ObservedGeneration
+// that is bigger or equal to the ObservedGeneration of the provided CertificateCondition.
+// Only the Type, Status and ObservedGeneration field will be used in the comparison,
+// meaning that this function will return 'true' even if the Reason, Message and
```

Comment:
In the codebase, ObservedGeneration property of CertificateCondition is always left as 0. This means that the condition will always be true in this case.
I need this extra condition, because I want to check if a Certificate that was in a Ready = True state, is still in Ready = True after changing the Certificate. Therefore, a Generation check is needed.
https://github.com/jetstack/cert-manager/blob/eeedc1c9bb6d28d7dc44ed8a9d53bfd2672bb61c/test/e2e/suite/conformance/certificates/tests.go#L722-L724

Reply:
It basically checks that the Certificate is newer than the CertificateCondition. Which is always true for the case where CertificateCondition is set to 0 (default).

Reply:
Thanks for explaining. I'd missed  or misread the "update" in the `WaitForCertificateReady*Update*`  function.